packages/lab/src/carousel/CarouselSlide.tsx

```diff
@@ -1,75 +1,62 @@
-import { makePrefixer } from "@salt-ds/core";
+import { SplitLayout, makePrefixer, useId } from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { clsx } from "clsx";
 import {
-  type ElementType,
   type HTMLAttributes,
-  type ReactElement,
+  type ReactNode,
   forwardRef,
+  useEffect,
   useRef,
 } from "react";
-
-import { clsx } from "clsx";
+import { useCarousel } from "./CarouselContext";
+import carouselSlideCss from "./CarouselSlide.css";
 
 export interface CarouselSlideProps extends HTMLAttributes<HTMLDivElement> {
-  ButtonBar?: ElementType;
-  Media: ReactElement;
-  description?: string;
-  title?: string;
-  contentAlignment?: "center" | "left" | "right";
+  /**
+   * Actions to be displayed in the content footer
+   */
+  actions?: ReactNode;
+  /**
+   * The image to be displayed inside the slide
+   */
+  media?: ReactNode;
 }
 
 const withBaseName = makePrefixer("saltCarouselSlide");
 
 export const CarouselSlide = forwardRef<HTMLDivElement, CarouselSlideProps>(
-  function CarouselSlide(
-    { ButtonBar, Media, description, title, contentAlignment },
-    ref,
-  ) {
-    const buttonBarRef = useRef(null);
+  function CarouselSlide({ actions, media, children }, ref) {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-carousel-slide",
+      css: carouselSlideCss,
+      window: targetWindow,
+    });
+
+    const slideRef = useRef(useId());
+    const { activeSlide, registerSlide, slides, bordered } = useCarousel();
+
+    useEffect(() => {
+      slideRef.current && registerSlide(slideRef.current);
+    }, [registerSlide]);
 
+    const isActive = slides[activeSlide] === slideRef.current;
     return (
-      <div ref={ref}>
-        {Media && <div className={withBaseName("mediaContainer")}>{Media}</div>}
-        <div className={withBaseName("fixedContainer")} ref={buttonBarRef}>
-          <div
-            className={clsx({
-              [withBaseName("textContainer")]: contentAlignment === "center",
-              [withBaseName("textContainerLeft")]: contentAlignment === "left",
+      <div role="group" ref={ref} className={withBaseName()}>
+        {media}
+        <div
+          className={clsx(withBaseName("content"), {
+            [withBaseName("bordered")]: bordered,
+          })}
+        >
+          {children}
+          <SplitLayout
```

Comment:
It's fine to use layout components in explorations but I think we should rely on CSS in our components